apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTest.cs

```diff
@@ -81,49 +81,34 @@ public async Task RetryTest(RetryTest test)
+    private async Task RunTestCaseAsync(InstructionList instructionList, Method method, bool expectSuccess, bool specifyPreconditions)
     {
         var context = CreateTestContext(method);
-        var response = await CreateRetryTestResource(instructionList, method);
-        try
+        var response = await CreateRetryTestResourceAsync(instructionList, method);
+
+        if (expectSuccess)
         {
-            if (expectSuccess)
+            RunRetryTest(response, context, method.Group, specifyPreconditions);
+            var postTestResponse = await GetRetryTestAsync(response.Id);
+            Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
+        }
+        else
+        {
+            try
             {
                 RunRetryTest(response, context, method.Group, specifyPreconditions);
-                var postTestResponse = await GetRetryTest(response.Id);
-                Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
-            }
-            else
-            {
-                try
-                {
-                    RunRetryTest(response, context, method.Group, specifyPreconditions);
 
-                    // storage.buckets.setIamPolicy has no preconditions implemented in .NET and hence will retry successfully
-                    if (!method.Name.Contains("buckets.setIamPolicy"))
-                    {
-                        Assert.False(response.Completed);
-                    }
-                }
-                catch (GoogleApiException ex) when (InstructionContainsErrorCode(ex.HttpStatusCode))
+                // storage.buckets.setIamPolicy has no preconditions implemented in .NET and hence will retry successfully
+                if (!method.Name.Contains("buckets.setIamPolicy"))
                 {
-                    // The instructions specified that the given status code would be returned.
-                    // We just need to check that we weren't expecting this specific call to succeed.
-                    Assert.False(expectSuccess);
+                    Assert.False(response.Completed);
                 }
             }
-        }
-        finally
-        {
-            try
+            catch (GoogleApiException ex) when (InstructionContainsErrorCode(ex.HttpStatusCode))
             {
-                DeleteStorageResources(context);
-                RemoveRetryIdHeader();
-                await DeleteRetryTest(response.Id);
-            }
-            catch
-            {
-                // Swallow any exceptions during clean-up: it's best effort.
+                // The instructions specified that the given status code would be returned.
+                // We just need to check that we weren't expecting this specific call to succeed.
+                Assert.False(expectSuccess);
```

Comment:
This code is redundant here, expectSuccess will never be true at this point, we are in  the else of `if(expectSuccess)`.